test/utils/url.spec.ts

```diff
@@ -72,15 +78,91 @@ describe("Url utils tests", () => {
   });
 
   describe("isRemoveJourney tests", () => {
-    test("returns true if query param journey=remove", () => {
+
+    test("returns true if app data not present in session and query param journey=remove", () => {
+      mockGetApplicationData.mockReturnValueOnce(undefined);
+
       req["query"] = {
         "journey": "remove"
       };
+
+      const result = urlUtils.isRemoveJourney(req);
+
+      expect(result).toBeTruthy();
+    });
+
+    test("returns true if is_remove is undefined in session data and query param journey=remove", () => {
+      mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
+
+      req["query"] = {
+        "journey": "remove"
+      };
+
+      const result = urlUtils.isRemoveJourney(req);
+
+      expect(result).toBeTruthy();
+    });
+
+    test("returns true if is_remove is null in session data and query param journey=remove", () => {
+      mockGetApplicationData.mockReturnValueOnce(
+        { ...APPLICATION_DATA_MOCK,
+          is_remove: null
+        }
+      );
+
+      req["query"] = {
+        "journey": "remove"
+      };
+
+      const result = urlUtils.isRemoveJourney(req);
+
+      expect(result).toBeTruthy();
+    });
+
+    test("returns true if is_remove is true in session data and query param journey=register", () => {
+      mockGetApplicationData.mockReturnValueOnce(
+        { ...APPLICATION_DATA_MOCK,
+          is_remove: true
+        }
+      );
+
+      req["query"] = {
+        "journey": "register"
+      };
+
       const result = urlUtils.isRemoveJourney(req);
 
       expect(result).toBeTruthy();
     });
 
+    test("returns true if is_remove is true in session data and query param journey not defined", () => {
+      mockGetApplicationData.mockReturnValueOnce(
+        { ...APPLICATION_DATA_MOCK,
+          is_remove: true
+        }
+      );
+
+      const result = urlUtils.isRemoveJourney(req);
+
+      expect(result).toBeTruthy();
+    });
+
+    test("returns false if is_remove is false in session data and query param journey=remove", () => {
```

Comment:
Shouldn't isRemoveJourney() return true it the remove query param exists?

Reply:
So this situation shouldn't arise - a conflict between what the `is_remove` value says and the `journey` query param value. But when discussed with @markpit and @mattch1 we thought that the `is_remove` value should take precedence.

Reply:
ok, makes sense. It would return true if the is_remove value is not present but the query param exists. I can see that's covered in another test.